### PR TITLE
Feature gate the `rustdoc::missing_doc_code_examples` lint

### PR DESCRIPTION
**Pull request by Nemo157**
Moves the lint from being implicitly active on nightly `rustdoc` to requiring a feature to activate, like other unstable lints.

Uses the new tracking issue https://github.com/rust-lang/rust/issues/101730

### CODE REVIEW

File: compiler/rustc_feature/src/active.rs [hunk 1]
```diff
@@ -221,6 +221,8 @@ declare_features! (
     (active, rustc_private, "1.0.0", Some(27812), None),
     /// Allows using internal rustdoc features like `doc(primitive)` or `doc(keyword)`.
     (active, rustdoc_internals, "1.58.0", Some(90418), None),
+    /// Allows using the `rustdoc::missing_doc_code_examples` lint
+    (active, rustdoc_missing_doc_code_examples, "1.31.0", Some(101730), None),
```
Comment (Nemo157) [inline compiler/rustc_feature/src/active.rs:225]: This is the version the original lint merged into, I'm not sure if this should just use the current nightly version instead though?

Comment (GuillaumeGomez) [reply compiler/rustc_feature/src/active.rs:225]: I suppose it doesn't matter much. The important version is the stable one once it gets stabilized.

**Comment by GuillaumeGomez** [comment]
> Uses the new tracking issue https://github.com/rust-lang/rust/pull/101732

This is the link for the current PR. Did you mean #101730?

**Comment by Nemo157** [comment]
Yes 🙄

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me, thanks! r=me once Ci pass

**Pull request merged**